src/mqttproto/_types.py

```diff
@@ -1088,9 +1106,15 @@ def decode(
         cls, data: memoryview, flags: int
     ) -> tuple[memoryview, MQTTPublishCompletePacket]:
         # Decode the variable header
+        reason_code = ReasonCode.SUCCESS
```

Comment:
I don't think we should allow the variable header to be empty in packets other than `DISCONNECT`, given that in the case of `DISCONNECT` the spec explicitly allows it, but not so for `PUBREC` or `PUBCOMP`.

Reply:
In the case of `PUB`* the variable header is not empty, in fact it cannot be because the message ID goes there. It's just shorter than usual because the broker skips the empty properties and the zero resultcode. MQTT section 3.4.2.1 explicitly allows the sender to do that.

In any case we can't decide not to accept this as it would make our client incompatible with several widely-used brokers.

Reply:
I looked at the spec again, and you're right. We should add tests for such packets.